frontend/lib/qr_code_scanner/qr_code_scanner.dart

```diff
@@ -180,41 +180,34 @@ class _QRViewState extends State<QRCodeScanner> {
     });
   }
 
-  void _tryParseCodeContent(String codeContent) {
-    try {
-      final cardDetails = parseQRCodeContent(codeContent);
-      // TODO this method gets called multiple times in a row for some reason
-      // The following code should be exclusive, but dart does not natively
-      // support mutexes, synchronized, etc.
-      if (isDone) {
-        return;
-      }
-      isDone = true;
-      Provider.of<CardDetailsModel>(context, listen: false)
-          .setCardDetails(cardDetails);
-      Navigator.of(context).maybePop();
-    } on Exception catch (e) {
+  void _tryParseCodeContent(String codeContent) async {
+    var wasSuccessful = false;
+    // needed because this method gets called multiple times in a row after one
+    // qr code gets detected, therefore we need to protect it
+    if (isDone) {
+      return;
+    }
+    isDone = true;
```

Comment:
Hm, still looks strange, but we already talked about it and I believe you that we need it 🤷🏽‍♂️

Reply:
I guess, otherwise the `_tryParseCodeContent` would be called several times and one after another waiting for the mutex and the doing their stuff with it...
If so, maybe call it `isWorking` or so... 

I guess, we don't need the mutex though... The mutex library also isn't really popular...

Reply:
I decided to use the mutex after @elKei24 added the mutex library for a different problem anyway. Probably could be removed, but I dont want to tinker with that right now because it works